### ISSUE TITLE
Make gopclntab search more flexible

### ISSUE THREAD

**Issue by d-we** [d-we/binja-golang-symbol-restore#1]
search in a more aggressive way to find gopclntab even if it is not at the beginning of a section. This could be useful for obfuscated binaries or binaries extracted from a coredump

**Comment by d-we** [comment]
we can do this by searching for the magic bytes and validating the first few table entries once we found matching magic bytes

**Comment by psifertex** [comment]
Related: https://cujo.com/reverse-engineering-go-binaries-with-ghidra/ has some discussion about the various sections/structures.

**Comment by d-we** [comment]
@psifertex thanks, I'll have a look in the next days

**Comment by psifertex** [comment]
FYI, I noticed that this plugin: https://github.com/f0rki/bn-goloader seems to handle obfuscated sections better.

I modified it to work with python3: https://github.com/psifertex/bn-goloader

But really, both that plugin and this are so similar that it would be nice if there was only one in the plugin manager that had the best of both.

**Comment by d-we** [comment]
@psifertex do you have an example binary where the other plugin performs better? If so, I could try to improve this one.
But if you think the other just performs better in any way, we can also remove this plugin from the plugin manager

**Comment by psifertex** [comment]
I've been talking to the author of https://go-re.tk/ and I think we can just use that as the basis for a BN plugin. Had to make sure the AGPLv3 license wasn't an issue, but given that, I think that's probably the better way to move forward. That way the part that handles all the go-specific stuff can be shared amongst other projects too and we can just use a thin wrapper around it for the BN side.

I worked on such a plugin on the live stream last friday so I think that'll probably be the best way forward. That said, that plugin will require some new APIs about to land on dev, so at least until we have a new stable there will be a reason to leave the other plugins in the repo.

TL;DR -- you shouldn't have to do anything and once we deprecate it we should have a good general solution going forward!
